src/main/java/org/lmdbjava/Dbi.java

```diff
@@ -186,12 +190,28 @@ public boolean delete(final Txn<T> txn, final T key, final T val) {
    * @param txn transaction handle (not null; not committed; must be R-W)
    */
   public void drop(final Txn<T> txn) {
+    drop(txn, false);
+  }
+
+  /**
+   * Drops the database. If delete is set to true, the database will be deleted
+   * and handle will be closed. See {@link #close()} for implication of handle close.
+   * Otherwise, only the data in this database will be dropped.
+   *
+   * @param txn transaction handle (not null; not committed; must be R-W)
+   * @param delete whether database should be deleted.
+   */
+  public void drop(final Txn<T> txn, final boolean delete) {
     if (SHOULD_CHECK) {
       requireNonNull(txn);
       txn.checkReady();
       txn.checkWritesAllowed();
     }
-    checkRc(LIB.mdb_drop(txn.pointer(), ptr, 0));
+    if (delete) {
+      clean();
+    }
+    final int del = delete ? 1 : 0;
+    checkRc(LIB.mdb_drop(txn.pointer(), ptr, del));
   }
```

Comment:
I'm unsure I agree that `drop(Txn, true)` should deallocate the comparator keys. The code as shown will deallocate the comparator keys, but this will be reattempted if the user separately invokes `close()`. 

As such either close state needs to be tracked (extra field), the comparator keys need to remove the `final` modifier so we can assign them to `null` (unusual within our `final`-preferring code base), or we simply make it the user's responsibility to invoke `close()` themselves if they wish to release the handles.

While `Txn` offers some precedent for us trying to track state and help the user avoid illegal use, `Txn` is a central API and frequently used. On the other hand, dropping and deleting a DBI is very unusual and I think it reasonable to expect users doing this consider the restrictions it involves (eg making sure no other thread or process is using the DBI, ensuring the now-defunct `Dbi` is no longer accessed once dropped, calling `Dbi.close()` if recovering the small amount of memory concerned out-weights the corruption risks noted in [mdb_dbi_close](http://www.lmdb.tech/doc/group__mdb.html#ga52dd98d0c542378370cd6b712ff961b5) etc).